File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length;
        int max=0;
        if(n==1){
            return true;
        }
        for(int i=0 ; i < n-1;i++){
            int sum=i+nums[i];
            max=Math.max(sum,max);

            if(i>=max){
                return false;
            }
            
        }
    return true;
    }
}

File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        i=0
        jump=0
        sum=0
        for num in nums:
            if jump<i:
                return False
            sum=i+num
            jump=max(sum,jump)
            
            i+=1
        return True

        